From 7b81933913dd0b282b36bcd8e82deb93b6caaca1 Mon Sep 17 00:00:00 2001
From: Wentao MA <wentao.ma@amlogic.com>
Date: Thu, 23 Dec 2021 18:30:17 +0800
Subject: [PATCH] libdvr: Playing timeshift lead to black screen [1/1]

PD#SWPL-68480

Problem:
On a newly flashed box, playing timeshift cause black screen.

Solution:
Fix it by initializing uninitialized variables.

Verify:
Locally tested ok in AH212 environment.

Signed-off-by: Wentao.MA <wentao.ma@amlogic.com>
Change-Id: I7f07de0ac481764a33fec8810eaf0319c69da350
---
 src/dvr_wrapper.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/dvr_wrapper.c b/src/dvr_wrapper.c
index 42564a5..e22689d 100644
--- a/src/dvr_wrapper.c
+++ b/src/dvr_wrapper.c
@@ -1425,12 +1425,12 @@ int dvr_wrapper_close_playback (DVR_WrapperPlayback_t playback)
 int dvr_wrapper_start_playback (DVR_WrapperPlayback_t playback, DVR_PlaybackFlag_t flags, DVR_PlaybackPids_t *p_pids)
 {
   DVR_WrapperCtx_t *ctx;
-  int error;
+  int error=0;
   uint64_t *p_segment_ids;
   uint32_t segment_nb;
   uint32_t i;
   DVR_RecordSegmentInfo_t seg_info_1st;
-  int got_1st_seg;
+  int got_1st_seg=0;
   DVR_WrapperCtx_t *ctx_record;/*for timeshift*/
   DVR_Bool_t is_timeshift = DVR_FALSE;
 
-- 
2.29.2

